pex/pex.py

```diff
@@ -619,21 +620,22 @@ def execute_script(self, script_name):
             return "Could not find script {!r} in pex!".format(script_name)
 
         TRACER.log("Found script {!r} in {!r}.".format(script_name, dist_script.dist))
-        if dist_script.is_python_script():
-            return self.execute_content(
+        ast = dist_script.python_script()
+        if ast:
+            return self.execute_ast(
                 dist_script.path, dist_script.read_contents(), argv0=script_name
             )
         else:
             return self.execute_external(dist_script.path)
 
     @staticmethod
-    def execute_external(cmd):
+    def execute_external(binary):
         # type: (str) -> Any
+        args = [binary] + sys.argv[1:]
         try:
-            proc = Executor.open_process([cmd] + sys.argv[:1])
-            return proc.wait()
+            return Executor.open_process(args).wait()
         except Executor.ExecutionError as e:
-            return "Could not invoke script: {}".format(e)
+            return "Could not invoke script {}: {}".format(binary, e)
```

Comment:
Did you see a error message here, where the binary name was missing?
I did not add a test for it, but I think that the error message I noticed included the name of the executable in the error message, now resulting in duplicated information in the error message.
However, I'm not sure if this will always be the case..

Reply:
Yeah, I'm not sure of that either. So, at the risk of duplicated information, I wanted to stuff what we know in their in case the layers down to the OS involved just say something like "file not found".

Reply:
To answer your question - I didn't see anything that directly prompted this change. The real prompt was the bug fixed above on line 634 where args were not being passed through correctly (`s/[:1]/[1:]/`). That ade me revisit this function wholesale.

Reply:
Oh, doh. Good catch.